### DISCUSSION TITLE
Hubs Cloud, it does not allow me to insert my room in an iframe

### DISCUSSION THREAD

**Discussion by Adrian-Serna** [mozilla/hubs#4265] [Hubs Cloud Questions] [answered]
Hi, I'm trying to insert a room from my cloud hub instance into an iframe from an external domain that I also manage.

I have modified the HTTP headers Content-Security-Policy: frame-ancestors to fix the first error I got:

`frame has been denied by the directive "X-Frame-Options" set to "SAMEORIGIN"`

This worked, but now a new error is called explicitly from the application code.

```
Error: Uncaught Error: no embed token
    at Module.1687 (hub.js: 190)
```

The code snippet that produces the error is this:

```
const qs = new URLSearchParams(location.search);
const isMobile = AFRAME.utils.device.isMobile();
const isMobileVR = AFRAME.utils.device.isMobileVR();
const isEmbed = window.self !== window.top;
if (isEmbed && !qs.get("embed_token")) {
  // Should be covered by X-Frame-Options, but just in case.
  throw new Error("no embed token");
}
```

What is the correct way in which I should approach this problem and allow the execution of one of my rooms within an iframe of my external domain?

Would it be enough to create my custom client eliminating this error call or am I missing something else?

I appreciate any information, thank you very much for your help.

**Comment by rawnsley** [comment, marked as answer]
The URL needs an embed code, which should be generated by the Invite menu. Here is an example:

`<iframe src="https://hubs.mozilla.com/AJeyeCM?embed_token=null" style="width: 1024px; height: 768px;" allow="microphone; camera; vr; speaker;"></iframe>`

**Reply by rawnsley** [reply]
Hmmm... this is a bad example because the code isn't been generated properly. Hopefully you will have more luck on your Hubs Cloud

**Reply by Adrian-Serna** [reply]
It was so surprisingly easy! Yes, it actually worked as expected. You don't even need HTTP Content-Security-Policy.
Thank you!

**Reply by Softvision-GeluHaiduc** [reply]
Marking this as answered, unmark the answer if you need any more information. Thank you!
